### DISCUSSION TITLE
Example 'object_detection' - compilation problem

### DISCUSSION THREAD

**Discussion by tpatora** [AxisCommunications/acap-native-sdk-examples#121] [General]
Hi,
On Mac when compile I've get a strange issue (below). Finally Docker Desktop crash after compilation error.
Any Idea?

object-detection % docker build --build-arg ARCH=armv7hf  --build-arg CHIP=chip --tag obj_detect:1.0 .
[+] Building 77.5s (11/22)                                                                                                          
 => [internal] load build definition from Dockerfile                                                                           0.0s
 => => transferring dockerfile: 2.46kB                                                                                         0.0s
 => [internal] load .dockerignore                                                                                              0.0s
 => => transferring context: 2B                                                                                                0.0s
 => [internal] load metadata for docker.io/axisecp/acap-native-sdk:1.7-armv7hf-ubuntu22.04                                     2.1s
 => [auth] axisecp/acap-native-sdk:pull token for registry-1.docker.io                                                         0.0s
 => [ 1/17] FROM docker.io/axisecp/acap-native-sdk:1.7-armv7hf-ubuntu22.04@sha256:0438aa946616c7ab9cfbe3a8c5a56873c9f4843b7d6  0.0s
 => [internal] load build context                                                                                              0.0s
 => => transferring context: 53.81kB                                                                                           0.0s
 => CACHED [ 2/17] WORKDIR /opt/build/libjpeg                                                                                  0.0s
 => CACHED [ 3/17] WORKDIR /opt/build                                                                                          0.0s
 => CACHED [ 4/17] RUN apt-get update && apt-get install --no-install-recommends -y cmake &&     apt-get clean &&     rm -rf   0.0s
 => CACHED [ 5/17] RUN git clone --branch 2.0.6 https://github.com/libjpeg-turbo/libjpeg-turbo.git                             0.0s
 => CACHED [ 6/17] WORKDIR /opt/build/libjpeg-turbo/build                                                                      0.0s
 => [ 7/17] RUN if [ "armv7hf" = armv7hf ]; then         gCFLAGS=' -O2 -mthumb -mfpu=neon -mfloat-abi=hard -mcpu=cortex-a9 -  75.4s
 => => # [ 74%] Building C object CMakeFiles/turbojpeg.dir/rdppm.c.o                                                               
 => => # [ 75%] Building C object CMakeFiles/turbojpeg.dir/jdarith.c.o                                                             
 => => # [ 73%] Building C object sharedlib/CMakeFiles/jpeg.dir/__/jquant2.c.o                                                     
 => => # [ 75%] Building C object CMakeFiles/turbojpeg.dir/jfdctfst.c.o                                                            
 => => # [ 75%] Building C object sharedlib/CMakeFiles/jpeg.dir/__/jcarith.c.o                                                     
 => => # [ 75%] Building C object sharedlib/CMakeFiles/jpeg.dir/__/jquant1.c.o                                                     
invalid character 'c' looking for beginning of value_

**Comment by pataxis** [comment]
Hi @tpatora 

Make sure that you have tested in a clean environment:
- clone a fresh repo to a separate directory
- make no changes to object-detection
- then build again
Do you still get the issue?
If so, can you run `docker system info` and paste the output or at least at least which Docker Desktop, Docker Engine, buildx version you have. Would also be good to know which OS version you are using.

Your output fail on step 7/17 but I also see that you have wrong CHIP and will fail on a later step 14/17. 
you need to match the SDK architecture with a correct chip, see the [README](https://github.com/AxisCommunications/acap-native-sdk-examples/tree/main/object-detection) for possible options.
